language/documentation/book/src/global-storage-operators.md

```diff
@@ -10,14 +10,14 @@ Move programs can create, delete, and update [resources](./structs-and-resources
 |`borrow_global<T>(address): &T`	             | Return an immutable reference to the `T` stored under `address`    | If `address` does not hold a `T`         |
 |`exists<T>(address): bool`                   | Return `true` if a `T` is stored under `address`                   |  Never                                      |
 
-Each of these instructions is parameterized by a resource type `T` declared *in the current module*. This ensures that a resource can only be manipulated via the API exposed by its defining module. The instructions also take either an [`address`](./address.md) or [`&signer`](./signer.md) representing the account address where the resource of type `T` is stored.
+Each of these instructions is parameterized by a type `T` with ability declared *in the current module*. This ensures that a resource can only be manipulated via the API exposed by its defining module. The instructions also take either an [`address`](./address.md) or [`&signer`](./signer.md) representing the account address where the resource of type `T` is stored.
```

Comment:
This doesn't parse very well for me: "parameterized by a type `T` with ability declared in the current module". I don't think you mean that the ability is declared in the current module. It also seems like "with ability" is too terse to be clear -- I don't understand anyway. Are you saying that each operation requires certain abilities?

Reply:
I think I just started typing and stopped :P 

I think it should read 

> Each of these instructions is parameterized by a type `T` (with the [`key` ability](./abilities.md)) declared *in the current module*.

Reply:
Or maybe

```
Each of these instructions is parameterized by a type `T` with the [`key` ability](./abilities.md). However, each type `T` *must be declared in the current module*. 
```